### ISSUE TITLE
Bulk matching job not showing results

### ISSUE THREAD

**Issue by mdoering** [Sp2000/colplus-frontend#1339]
When I run a bulk matching job by uploading a CSV file I do get a successful email notification, but the UI stays with a rolling ball on the job page and does not show a finished result for download

**Comment by thomasstjerne** [comment]
The UI is a bit hacky here because a job "disaapears" when its finished. So the UI does polling, but suddenly the job id gives a 404. Then the UI does a head request on e.g. https://api.checklistbank.org/job/a6045d10-93f3-4651-b76d-8f22d4ea180b.zip but this now seems to redirect to https://download.checklistbank.org/job/a6/a6045d10-93f3-4651-b76d-8f22d4ea180b.zip which is blocked by CORS blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.  It would be nice if the matching jobs behaved like downloads, i.e. don´t disappear but mark as finished or failed

**Comment by mdoering** [comment]
Yes, all background jobs should really behave the same in the basics and be stored.
That needs quite a bit of refactoring on both sides though
